### PR TITLE
Remove duplicate input_node() methods

### PR DESCRIPTION
**Pull request by phate**
We had three incarnations of the same method for retrieving the node connected to an input. Remove two, keep one.

### CODE REVIEW

File: jlm/llvm/opt/push.cpp [hunk 1]
```diff
@@ -185,7 +185,7 @@ push(jlm::rvsdg::gamma_node * gamma)
       auto argument = region->argument(n);
       for (const auto & user : *argument)
       {
-        auto tmp = input_node(user);
+        auto tmp = jlm::rvsdg::input::GetNode(*user);
```
Comment (haved) [inline jlm/llvm/opt/push.cpp:188]: It seems only this file has the `jlm::` prefix included in all the replaced calls. Is there something special about this file or just a simple mistake?

**Review by haved — APPROVED** [review]
Looks good, just one comment

File: jlm/llvm/opt/push.cpp (hunk 1 above)
Comment (phate) [reply jlm/llvm/opt/push.cpp:188]: The functions are not in the jlm namespace (yet). This file has not been properly refactored yet. That is why the `jlm::` prefix was needed.

**Review by caleridas — APPROVED** [review]
Looks good

**Pull request merged**